File: 2020/day8/part1.js
const path = require('path')
const readFile = require('../../utils/read-file')

const boot = async () => {
  const data = await readFile(path.resolve(__dirname, 'data.txt'))
  const commands = data.toString().trim().replace(/\+/g, '').split('\n').map(x => x.split(' '))
  const accumulator = run(commands)
  console.log(accumulator)
}

const run = (commands) => {
  const stack = []
  let accumulator = 0
  let noDuplicates = true
  let i = 0
  do {
    if (stack.includes(commands[i])) {
      noDuplicates = false
    } else {
      stack.push(commands[i])
      switch (commands[i][0]) {
        case 'acc':
          accumulator += Number(commands[i][1])
          i += 1
          break
        case 'jmp':
          i += Number(commands[i][1])
          break
        default:
          i += 1
          break
      }
    }
  } while (noDuplicates)

  return accumulator
}

(async function () {
  await boot()
}())

File: 2021/day2/part_1.py
import os
f = open(os.path.join(os.path.dirname(__file__), 'data.txt'))
commands = f.read().splitlines()

depth = 0
horizontal = 0

for command in commands:
    commandSet = command.split(' ')
    if commandSet[0] == 'up':
        depth -= int(commandSet[1])
    elif commandSet[0] == 'down':
        depth += int(commandSet[1])
    else:
        horizontal += int(commandSet[1])

print(depth * horizontal)
